web3/utils/filters.py

```diff
@@ -106,7 +116,27 @@ def is_valid_entry(self, entry):
         """
         return True
 
+    def _filter_valid_entries(self, entries):
+        return filter(
+            lambda entry: self.is_valid_entry(entry),
```

Comment:
I think this lambda is not necessary.  You should be able to do `return filter(self.is_valid_entry, entries)`